src/control/Control.Layers.js

```diff
@@ -342,6 +343,7 @@ export var Layers = Control.extend({
 		// https://github.com/Leaflet/Leaflet/issues/2771
 		var holder = document.createElement('div');
 
+		label.setAttribute('for', 'layer-select');

```

Comment:
The `for` attribute is specifically for when you need to associate an `input` that is not a child element of `label`. I suggest reverting changes in Control.Layers.js. Other than that LGTM!

Reply:
Oh okay. Thanks for the feedback! @Malvoz 
I have reverted these changes as per your suggestions.
Let me know if anything else needs to be changed.
Thanks!